### ISSUE TITLE
Should not we store the database file in `/data` by default?

### ISSUE THREAD

**Issue by felipecrs** [blakeblackshear/frigate#4856]
1. `/data` is unique per add-on, which persists for as long as the add-on is installed
2. Uninstalling the add-on would cause the database file to get deleted, which I think makes sense
3. I suppose several people maps their `/media/frigate` folder to an external (potentially an much slower HDD) device. It isn't smart to store the database in the HDD in this case.

**Comment by NickM-27** [comment]
I'm not sure I see a good way to change this without being a major breaking change, I agree anyone with /media/frigate would want the db on a separate drive, but it is a complicated thing to support existing users

**Comment by felipecrs** [comment]
How about:

- If a custom db path is set in frigate.yml, do nothing, use it as is
- Otherwise, check if `/media/frigate/frigate.db` (or whatever is the default filename) exists. If it does, move it to `/data/frigate.db`.

**Comment by NickM-27** [comment]
that only works for the addon though which has /data/ passed to it. Any non addon user likely doesn't have /data/ mapped which means all the future DB changes would not be saved once frigate restarted and the db would be gone

**Comment by felipecrs** [comment]
Docs can be updated to include this additional volume mount. Also, this can be mentioned as part of the release notes. Something like:

- The database is now by default stored at /data. Please add this additional volume mount to your docker-compose.yaml to ensure it will persist across restarts.

---

Or we can of course use different database paths by default for add-on vs docker. This can be handled during add-on startup. Something like:

```
if "running as addon"; then
  echo '/data/frigate.db' > /var/run/s6/container_environment/FRIGATE_DEFAULT_DB_PATH
fi
```

And then, Frigate as Python can read the `FRIGATE_DEFAULT_DB_PATH` environment variable. But I still prefer the first option. Chances are that even people running as a container will still map /media to a separate disk.

**Comment by NickM-27** [comment]
the problem is not everyone reads the release notes, in general it is what it is, but a breaking change which would result in data lost if the user didn't read the release notes is not acceptable IMO.

I also have a problem with

> Uninstalling the add-on would cause the database file to get deleted, which I think makes sense

many users switch between the normal and full access addon and between the normal and beta addon, we don't want that operation to result in lost data

**Comment by felipecrs** [comment]
In this case, `/config` is a better place than `/media`.

**Comment by felipecrs** [comment]
But I'm all in for breaking changes as Frigate is still in 0.x. This is in accordance with SemVer, which expects breaking changes to occur in development phase.

**Comment by felipecrs** [comment]
Another option would be to store the db at `/media/frigate.db` instead of `/media/frigate/frigate.db`, supposing people would map `/media/frigate` to an external disk rather than the whole `/media`.

**Comment by NickM-27** [comment]
> But I'm all in for breaking changes as Frigate is still in 0.x. This is in accordance with SemVer, which expects breaking changes to occur in the development phase.

That's fair, but as someone who works to support users in the issues it is not that simple and being in 0.x doesn't change the time it takes to help users solve the breaking changes. Also, we're trying to move away from breaking changes as it just makes things more difficult for everyone involved.

**Comment by blakeblackshear** [comment]
I think this change is probably a good idea. It just needs to be done very carefully.

**Comment by felipecrs** [comment]
Home Assistant stores its db in `/config`:

```console
> ls -lha /config/*.db
-rw-r--r--    1 root     root     1000.8M Jan 23 20:30 /config/home-assistant_v2.db
```

Thus, I'd suggest we do it as well. There's no downside of it, I think.

**Comment by NickM-27** [comment]
> Thus, I'd suggest we do it as well. There's no downside of it, I think.

many users right now have their config file mapped directly instead of the entire config direcory, so it would end up being written to nothing for non addon installs and when the container restarts the DB is erased with it.

**Comment by felipecrs** [comment]
Oh yeah. I was only thinking about the add-on users. Here's something we can do:

```bash
# db migration

if [[ there is nothing set in frigate.yaml:database.path ]]; then
  if [[ -f /media/frigate/frigate.db ]]; then
    if mountpoint /config; then
      # /config is a mount point, move the db
      mv -f /media/frigate/frigate.db
    else
      echo "[ERROR] Trying to move the db to /config but it's not a mountpoint, please mount the /config folder"
      exit 1
    fi
  fi
fi
```

**Comment by felipecrs** [comment]
![image](https://user-images.githubusercontent.com/29582865/214177045-877a62de-54b9-42a3-980f-1fda48e0683f.png)

**Comment by NickM-27** [comment]
yeah, that does seem like a solution that could work 🤔

**Comment by felipecrs** [comment]
Now... should it be done as part of the Frigate process, or should we do as a s6 oneshot/init service?

**Comment by NickM-27** [comment]
> Now... should it be done as part of the Frigate process, or should we do as a s6 oneshot/init service?

Feel free to work on it but just FYI I don't believe it will be included in 0.12. We are looking to eventually break different processes into their own S6 processes along with the potential idea for zeromq so I think it would make sense as an S6 service init
